src/utils.jl

```diff
@@ -180,5 +180,56 @@ makesym(t::Symbolic; kwargs...) = Sym{symtype(t)}(tosymbol(t; kwargs...))
 makesym(t::Num; kwargs...) = makesym(value(t); kwargs...)
 
 var_from_nested_derivative(x, i=0) = (missing, missing)
-var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1],i+1) : (x,i)
-var_from_nested_derivative(x::Sym,i=0) = (x,i)
+var_from_nested_derivative(x::Term,i=0) = operation(x) isa Differential ? var_from_nested_derivative(arguments(x)[1], i + 1) : (x, i)
+var_from_nested_derivative(x::Sym,i=0) = (x, i)
+
+function degree(p::Sym, sym::Union{Sym,Nothing}=nothing)
```

Comment:
Why does it have to be a `Sym` or nothing? Couldn't it be a term as well? Like `degree(sin(x)^2, sin(x))`?

Reply:
Makes sense, thanks for pointing that out. I will take a look a bit later.

Reply:
This is now added, for instance `degree(sin(x)^2+exp(y))` returns 2.